src/macros.rs

```diff
@@ -1,10 +1,305 @@
+//! Extra macro documentation.
+
 #[doc(hidden)]
 pub use fizyr_rpc_macros::interface as interface_impl;
 
 #[macro_export]
 /// Define an RPC interface.
+///
+/// This macro generates a module with a `Client` and `Server` struct,
+/// and some more helper types.
+///
+/// The client struct can be used to initiate requests and send stream messages.
+/// It can be created from a [`PeerWriteHandle`] or a [`PeerHandle`].
+/// Note that if you create the client from a [`PeerHandle`], the [`PeerReadHandle`] will not be accessible.
+///
+/// For the server struct it is exacatly opposite: it can be used to receive requests and stream messages.
+/// It can be created from a [`PeerReadHandle`] or a [`PeerHandle`],
+/// but creating it from a [`PeerHandle`] will discard the [`PeerWriteHandle`].
```

Comment:
Because servers can't initiate interactions, right?

Reply:
Well, that depends on your terminology. The listener can accept connections and then initiate requests. But  the generated Server can only handle requests and stream messages.